astropy/coordinates/representation.py

```diff
@@ -1837,10 +1837,15 @@ def transform(self, matrix):
         NotImplementedError
 
         """
-        raise NotImplementedError(
-            "Radial representations cannot be transformed "
-            "by a Cartesian matrix."
-        )
+        not_diagonal = np.any(np.count_nonzero(x - np.diag(np.diagonal(x))))
```

Comment:
`x` should be `matrix`.

But more importantly: this is not quite good enough: all diagonal elements need to be equal. So perhaps
```
scale = matrix[..., 0, 0]
if np.any(matrix != scale[..., np.newaxis, np.newaxis] * np.eye(3)):
    raise ...

return self * scale
```

Reply:
That's quite clean. I've implemented it.